lib/Parser/Hash.h

```diff
@@ -266,8 +267,16 @@ struct Ident
                     // Without parameter scope, we would have just pushed the ref instead of inserting.
                     // We effectively had a false positive match (a parameter scope ref with no sym)
                     // so we need to push the Pid rather than inserting.
-                    newRef->prev = m_pidRefStack;
-                    m_pidRefStack = newRef;
+                    if (prevRef)
+                    {
+                        newRef->prev = prevRef->prev;
+                        prevRef->prev = newRef;
+                    }
+                    else
+                    {
+                        newRef->prev = m_pidRefStack;
```

Comment:
Yes. if we have some code like function (a = x) { var x = 1; } then the prevRef will be null. So we can't get rid of the else part.